x-pack/plugins/security_solution/public/common/hooks/esql/use_esql_availability.ts

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useMemo } from 'react';
+import { useKibana } from '../../lib/kibana';
+import { useIsExperimentalFeatureEnabled } from '../use_experimental_features';
+
+export const useEsqlAvailability = () => {
```

Comment:
even though this hook is pretty straightforward to understand, if you want I would add a short documentation to describe what the hook does. Up to you!

Reply:
Thanks, will add a comment in the PR right after this! Hoping to make the next BC!